python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py

```diff
@@ -160,6 +183,42 @@ def run(
 
         return PipesClientCompletedInvocation(pipes_session)
 
+    def _enrich_submit_task_dict(
+        self, context: OpExecutionContext, session: PipesSession, submit_task_dict: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        if "existing_cluster_id" in submit_task_dict:
+            # we can't set env vars on an existing cluster
+            # so we must use CLI to pass Pipes params
+            cli_args = session.get_bootstrap_cli_arguments()  # this is a mapping
+
+            for task_type in self.get_task_fields_which_support_cli_parameters():
+                if task_type in submit_task_dict:
+                    existing_params = submit_task_dict[task_type].get("parameters", [])
+
+                    # merge the existing parameters with the CLI arguments
+                    for key, value in cli_args.items():
+                        existing_params.extend([key, value])
+
+                    submit_task_dict[task_type]["parameters"] = existing_params
+                    context.log.debug(
+                        f'Passing Pipes bootstrap parameters via Databricks parameters as "{key}.parameters". Make sure to use the PipesCliArgsParamsLoader in the task.'
+                    )
+                    break
+
+        else:
+            pipes_env_vars = session.get_bootstrap_env_vars()
+
+            submit_task_dict["new_cluster"]["spark_env_vars"] = {
+                **submit_task_dict["new_cluster"].get("spark_env_vars", {}),
+                **(self.env or {}),
+                **pipes_env_vars,
+            }
+
+        return submit_task_dict
+
+    def get_task_fields_which_support_cli_parameters(self) -> Set[str]:
+        return {"spark_python_task", "python_wheel_task"}
```

Comment:
Any reason this is a function instead of a constant

Reply:
There are also other task types which can potentially accept parameters, but I didn't want to spend time on supporting them now (before this PR we only really supported `spark_python_task`). 

This being a method makes it easier for users to customize the PipesClient.